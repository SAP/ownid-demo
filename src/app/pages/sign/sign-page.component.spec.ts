import { async, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { RouterTestingModule } from '@angular/router/testing';
import { Router } from '@angular/router';
import { IOwnidRs, SignPageComponent } from './sign-page.component';

describe('SignPageComponent', () => {
  let router: Router;

  beforeEach(() => {
    router = {} as Router;
    router.navigateByUrl = jest.fn();
  });

  describe('Snapshot', () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        imports: [RouterTestingModule],
        declarations: [SignPageComponent],
        schemas: [NO_ERRORS_SCHEMA],
      }).compileComponents();
    }));

    test('check snapshot', () => {
      const fixture = TestBed.createComponent(SignPageComponent);

      expect(fixture).toMatchSnapshot();
    });
  });

  describe('onSuccessLogin', () => {
    it('should redirect to /account and set cookies', () => {
      const sut = new SignPageComponent(router);
      const cookieName = 'fakeName';
      const cookieValue = 'fakeValue';
      sut.onSuccessLogin({ sessionInfo: { cookieName, cookieValue } } as IOwnidRs);
      // expect(router.navigateByUrl).toBeCalledWith('/account');
    });
  });
});
