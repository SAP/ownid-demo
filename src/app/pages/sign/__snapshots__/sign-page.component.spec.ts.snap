// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignPageComponent Snapshot check snapshot 1`] = `
<sign-page
  router={[Function Router]}
>
  <section
    class="sign-body"
  >
    <div
      class="container"
    >
      <div
        class="call-2-action"
      >
        <h1
          class="headline"
        >
           For the best experience use Own Id for login 
        </h1>
        <h2
          class="sub-title"
        >
           Scan QR code with your phone and proceed with out any logins and passwords 
        </h2>
      </div>
      <div>
        <mat-card
          class="sign-section"
        >
          <mat-tab-group
            animationduration="0ms"
            mat-stretch-tabs=""
          >
            <mat-tab
              label="Login"
            >
              <ownid
                type="login"
              />
              <login-form />
            </mat-tab>
            <mat-tab
              label="Register"
            >
              <registration-form />
            </mat-tab>
          </mat-tab-group>
        </mat-card>
      </div>
    </div>
  </section>
</sign-page>
`;
